### DISCUSSION TITLE
any way to take a look at the final kernel function?

### DISCUSSION THREAD

**Discussion by LeiWang1999** [NVIDIA/cutlass#602] [General]
Is there any way to take a look at the final kernel function that generated by the huge, complex cutlass template library?

hard to analysis : )

**Comment by hwu36** [comment]
You can use cuobjdump (https://docs.nvidia.com/cuda/cuda-binary-utilities/index.html#cuobjdump) to dump ptx or sass.  Is that what you want?

**Reply by LeiWang1999** [reply]
Thanks for reply!

However sass or ptx code is not the outputs that I expect, what I want is something like a cuda file contain a kernel that can have a clean view of the thread's behavior

**Reply by hwu36** [reply]
you can use `-keep -v` in `nvcc` command line to keep other intermediate files.

**Reply by LeiWang1999** [reply]
Thanks, I used the following command to build cutlass, while cmakeCache shows the intermediate file will be put on build/tmp directorty, but there is no files in the tmp dir,  any problems?
```bash
cmake ..
-DCUTLASS_NVCC_ARCHS=86
-DCUTLASS_LIBRARY_KERNELS=all
-DCUTLASS_LIBRARY_OPERATIONS=gemm
-DCMAKE_CUDA_COMPILER=/usr/local/cuda/bin/nvcc
-DCUTLASS_NVCC_VERBOSE=ON
-DCUTLASS_NVCC_KEEP=ON
```

**Reply by hwu36** [reply]
search `*.ptx` in your build directory, then you can find them.

BTW, don't use `all` in `DCUTLASS_LIBRARY_OPERATIONS`, you will build too many kernels.  Use the default one or regexp if you know what you want.
